crates/router/src/core/payment_methods/utils.rs

```diff
@@ -0,0 +1,857 @@
+                    None,
+                    None::<()>,
+                    None,
+                )
+                .map_err(KgraphError::GraphConstructionError)?;
+
+            agg_nodes.push((
+                card_and_node,
+                cgraph::Relation::Positive,
+                cgraph::Strength::Strong,
+            ));
+        } else if let Some(connector_list) =
+            supported_pm_for_mandates.0.get(&pmt.payment_method_type)
+        {
+            non_card_dir_values.extend(
+                connector_list
+                    .connector_list
+                    .clone()
+                    .into_iter()
+                    .filter_map(|connector| {
+                        api_enums::RoutableConnectors::from_str(connector.to_string().as_str())
+                            .ok()
+                            .map(|connector| {
+                                dir::DirValue::Connector(Box::new(
+                                    api_models::routing::ast::ConnectorChoice {
+                                        connector,
+                                        #[cfg(not(feature = "connector_choice_mca_id"))]
+                                        sub_label: None,
+                                    },
+                                ))
+                            })
+                    }),
+            );
+            let non_card_mandate_in_node = builder
+                .make_in_aggregator(non_card_dir_values, None, None::<()>)
+                .map_err(KgraphError::GraphConstructionError)?;
+
+            let non_card_and_node = builder
+                .make_all_aggregator(
+                    &[
+                        (
+                            card_value_node,
+                            cgraph::Relation::Negative,
+                            cgraph::Strength::Strong,
+                        ),
+                        (
+                            payment_type_value_node,
+                            cgraph::Relation::Positive,
+                            cgraph::Strength::Strong,
+                        ),
+                        (
+                            non_card_mandate_in_node,
+                            cgraph::Relation::Positive,
+                            cgraph::Strength::Strong,
+                        ),
+                    ],
+                    None,
+                    None::<()>,
+                    None,
+                )
+                .map_err(KgraphError::GraphConstructionError)?;
+
+            agg_nodes.push((
+                non_card_and_node,
+                cgraph::Relation::Positive,
+                cgraph::Strength::Strong,
+            ));
+        }
+    }
+
+    Ok(Some(
+        builder
+            .make_any_aggregator(
+                &agg_nodes,
+                Some("any node for card and non card pm"),
+                None::<()>,
+                None,
+            )
+            .map_err(KgraphError::GraphConstructionError)?,
```

Comment:
Do we need to construct an `any` aggregator here? Cuz from what I can see in this function, we will construct either a card specific subgraph, or a non-card-specific subgraph, but never both at once. So the `any` aggregator here will probably always have only 1 predecessor, making it redundant. Please check and correct me if I'm wrong.

Reply:
We can probably discuss on this and can refactor this in upcoming PR. If required can linkup an issue here as well. Anyways we have `update_mandate` flow supported by one connector(cybersource) only. So seeing the timelines if it is not necessary we can ignore it for this PR. We can discuss and change accordingly in the next PR.